colossus/src/main/scala/colossus/controller/Controller.scala

```diff
@@ -297,8 +300,11 @@ extends ConnectionHandler {
           }
           currentTrigger = Some(trigger)
         }
-        case Failure(reason) => {
-          //todo: what to do here?
+        //TODO: are the following 2 cases really exceptions..seems like they are part of the normal control flow from a controller's POV
```

Comment:
So..I'm not sure how convinced I am of this code.  If the pipe is closed/terminated, we should def reset our sink, as we don't want to push to it..but..there might be more needed here.  Def something to think through.  We might want to have some kind of pluggable behavior here..where if a pipe is closed/terminated while the controller is processing..maybe the controller can take some different actions: ignore, die, etc.  

As I was writing a test for this behavior..is when I started to doubt how "complete" just resetting the sink is.